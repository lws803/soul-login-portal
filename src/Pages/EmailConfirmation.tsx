import React, { useState, useEffect } from 'react';
import axios from 'axios';
import { Center, Spinner, Text } from '@chakra-ui/react';

import { BASE_URL } from '../constants';
import Page from '../components/Page';
import useQueryParams from '../hooks/useQueryParams';

export default function EmailConfirmation() {
  const [errors, setErrors] = useState<string[]>([]);
  const [loading, setLoading] = useState(false);

  const query = useQueryParams();

  useEffect(() => {
    setLoading(true);
    axios
      .post(
        `${BASE_URL}/v1/users/verify-confirmation-token?token=${query.get(
          'token',
        )}`,
      )
      .then(() => {
        setLoading(false);
      })
      .catch((error: any) => {
        setErrors([error.response.data.message]);
      });
  }, [query]);

  return (
    <Page errors={errors}>
      <Center mt={10}>
        {loading ? (
          <Spinner size="xl" />
        ) : (
          <Text fontSize="3xl" textAlign="center">
            Email confirmed!
          </Text>
        )}
      </Center>
    </Page>
  );
}
