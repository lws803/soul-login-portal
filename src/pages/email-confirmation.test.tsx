import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import axios from 'axios';

import EmailConfirmation from './email-confirmation';

describe(EmailConfirmation, () => {
  const path = '/?token=CODE';

  it('renders', async () => {
    const emailConfirmationCall = jest
      .spyOn(axios, 'post')
      .mockResolvedValue({});

    const { container } = render(
      <MemoryRouter initialEntries={[path]}>
        <EmailConfirmation />
      </MemoryRouter>,
    );

    expect(container).toMatchSnapshot();
    expect(emailConfirmationCall).toHaveBeenCalledWith(
      'https://api.soul-network.com/v1/users/verify-confirmation-token?token=CODE',
    );
  });
});
