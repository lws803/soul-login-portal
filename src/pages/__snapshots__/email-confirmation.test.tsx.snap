// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmailConfirmation renders 1`] = `
<div>
  <div
    class="css-0"
  >
    <div
      class="css-gmuwbf"
    >
      <div
        class="css-3qxem8"
      >
        <div
          class="css-1yfye7s"
        >
          <div
            class="chakra-spinner css-1y7joxr"
          >
            <span
              class="css-f8n5zr"
            >
              Loading...
            </span>
          </div>
        </div>
        <div
          class="chakra-stack css-untcqu"
        />
      </div>
    </div>
    <div
      class="css-gmuwbf"
    >
      <div
        class="chakra-stack css-zldu2j"
      >
        <a
          class="chakra-link css-0"
          href="https://github.com/soul-project/login-portal"
          target="_blank"
        >
          Github
        </a>
        <a
          class="chakra-link css-0"
          href="https://github.com/soul-project/soul"
          target="_blank"
        >
          Soul Network"s Github
        </a>
        <a
          class="chakra-link css-0"
          href="mailto:lws803@gmail.com"
          target="_blank"
        >
          Contact me
        </a>
      </div>
    </div>
  </div>
</div>
`;
